src/Fixer/ListNotation/ListSyntaxFixer.php

```diff
@@ -0,0 +1,163 @@
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class ListSyntaxFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    private $candidateTokenKind;
+    private $fixCallback;
+
+    /**
+     * Use 'syntax' => 'long'|'short'.
+     *
+     * @param array<string, string>|null $configuration
+     *
+     * @throws InvalidFixerConfigurationException
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->resolveCandidateTokenKind();
+        $this->resolveFixCallback();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'List (`array` destructuring) assignment should be declared using the configured syntax. Requires PHP >= 7.1.',
+            array(
+                new VersionSpecificCodeSample(
+                    "<?php\n[\$sample] = \$array;",
+                    new VersionSpecification(70100),
+                    array('syntax' => 'long')
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nlist(\$sample) = \$array;",
+                    new VersionSpecification(70100),
+                    array('syntax' => 'short')
+                ),
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before the BinaryOperatorSpacesFixer and TernaryOperatorSpacesFixer.
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70100 && $tokens->isTokenKindFound($this->candidateTokenKind);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $callback = $this->fixCallback;
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if ($tokens[$index]->isGivenKind($this->candidateTokenKind)) {
+                $this->$callback($tokens, $index);
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        $syntax = new FixerOptionBuilder('syntax', 'Whether to use the `long` or `short` `list` syntax.');
+        $syntax = $syntax
+            ->setAllowedValues(array('long', 'short'))
+            ->setDefault('long')
```

Comment:
sadly no, as all PSR's are made before PHP7 and were not amended, my guess is that for SF standard it will be long as it would create a lot of merging issues as long as they keep maintaining branches support PHP > 7.1